### PR TITLE
Commits discarded by history rewriting should be ignored and branch considered as new

### PR DESCRIPTION
**Pull request by antoine-malliarakis**
### Checklist

- [x] <del>All new jobs, commands, executors, parameters have descriptions</del> (not relevant here)
- [x] <del>Examples have been added for any significant new features</del> (not relevant here)
- [x] <del>README has been updated, if necessary</del> (not relevant here)

### Motivation, issues

Behaviours in case of branch being rebased or when the job previous runs included entries on commit which had been discarded (following a rebase or a squash) was... not working.

This is basically what's being described in the 3 below issues:
- https://github.com/iynere/compare-url/issues/41
- https://github.com/iynere/compare-url/issues/40
- https://github.com/iynere/compare-url/issues/25

### Description

Basically there was an issue which occurred with history rewrites (which can happen often in cases where people tend to squash their commits and / or rebase at head before merging, which is my case).

The idea is to basically say that:
* If a commit from a previous build was lost it should be ignored
* If a branch's previous job was on a no longer existing commit it should be considered as a new branch

### CODE REVIEW

**Comment by LoicEvenium** [comment]
This should fix #25 also I think
